.circleci/config.yml

```diff
@@ -80,8 +80,8 @@ jobs:
             . venv/bin/activate
             mkdocs build
   docs-deploy:
-    docker:
-      - image: python:3.5
+    machine:
```

Comment:
Did you test this? Why is this change needed?

Reply:
This has not been tested with this particular setup because I haven't had the opportunity to use it before. However, I consulted relevant documentation and verified it has been executed in the manner Circle desires.

As for why this change is required, the `docs-deploy` script now requires Node in addition to Python. This is due to the fact that the command metadata and docgen scripts are exempt from version control because they're not relevant to the operation of the app. As such they are non-persistent and must be run prior to GitHub Pages deployment (Which removes the ability to run the autogen scripts in a separate service). If the Python Docker image is used, Node is not available, and vice versa. Specifying Node as a secondary container will not work either as per CircleCI, since all build scripts run in the primary container.

The `machine` system, on the other hand, has several language tools pre-installed, among those Node and Python 3.5. In this case I saw fit to use `machine` despite it having a slightly longer startup time of approximately 30-60 seconds, which I think our workflow can absorb quite well.